glance_adoption.md

```diff
@@ -0,0 +1,240 @@
+
+Patch OpenStackControlPlane to deploy Glance:
+
+```
+oc patch openstackcontrolplane openstack --type=merge --patch '
+spec:
+  glance:
+    enabled: true
+    template:
+      databaseInstance: openstack
+      containerImage: quay.io/tripleozedcentos9/openstack-glance-api:current-tripleo
+      customServiceConfig: |
+        [DEFAULT]
+        enabled_backends=default_backend:rbd
+        [glance_store]
+        default_backend=default_backend
+        [default_backend]
+        rbd_store_ceph_conf=/etc/ceph/ceph.conf
+        rbd_store_user=openstack
+        rbd_store_pool=images
+        store_description=Ceph glance store backend.
+      storageClass: "local-storage"
+      storageRequest: 10G
+      glanceAPIInternal:
+        containerImage: quay.io/tripleozedcentos9/openstack-glance-api:current-tripleo
+      glanceAPIExternal:
+        containerImage: quay.io/tripleozedcentos9/openstack-glance-api:current-tripleo
+  extraMounts:
+    - extraVol:
+      - propagation:
+        - Glance
+        extraVolType: Ceph
+        volumes:
+        - name: ceph
+          projected:
+            sources:
+            - secret:
+                name: ceph-conf-files
+        mounts:
+        - name: ceph
+          mountPath: "/etc/ceph"
+          readOnly: true
+'
+```
+
+## Test the glance service from the OpenStack cli
+
+Inspect the resulting glance pods:
+
+```
+sh-5.1# cat /etc/glance/glance.conf.d/01-custom.conf
+
+[DEFAULT]
+enabled_backends=default_backend:rbd,ceph1:rbd
+enabled_backends=default_backend:rbd
+[glance_store]
+default_backend=default_backend
+[default_backend]
+rbd_store_ceph_conf=/etc/ceph/ceph.conf
+rbd_store_user=openstack
+rbd_store_pool=images
+store_description=Ceph glance store backend.
+
+sh-5.1# ls /etc/ceph/ceph*
+/etc/ceph/ceph.client.openstack.keyring  /etc/ceph/ceph.conf
+```
+
+Ceph secrets are properly mounted, at this point let's move to the openstack
+cli and check the service is active and the endpoints are properly updated.
+
+
+```
+(openstack)$ endpoint list | grep image
+
+| 569ed81064f84d4a91e0d2d807e4c1f1 | regionOne | glance       | image        | True    | internal  | http://glance-internal-openstack.apps-crc.testing   |
+| 5843fae70cba4e73b29d4aff3e8b616c | regionOne | glance       | image        | True    | public    | http://glance-public-openstack.apps-crc.testing     |
+| 709859219bc24ab9ac548eab74ad4dd5 | regionOne | glance       | image        | True    | admin     | http://glance-admin-openstack.apps-crc.testing      |
+```
+
+## Image upload
+
+We can test that an image can be created on from the adopted service.
+
+```
+(openstack)$ export OS_CLIENT_CONFIG_FILE=clouds-adopted.yaml
+(openstack)$ export OS_CLOUD=adopted
+(openstack)$ curl -L -o /tmp/cirros-0.5.2-x86_64-disk.img http://download.cirros-cloud.net/0.5.2/cirros-0.5.2-x86_64-disk.img
+    qemu-img convert -O raw /tmp/cirros-0.5.2-x86_64-disk.img /tmp/cirros-0.5.2-x86_64-disk.img.raw
```

Comment:
If you use interoperable image import's image_conversion plugin then this step is not required, we can enhance it later to include how to enable image conversion plugin.

Reply:
yeah, that's nice and good to follow up!